cli/src/commands/extensions/api.rs

```diff
@@ -51,8 +67,17 @@ async fn analyze(
         },
     };
 
+    let packages = packages
+        .drain(..)
```

Comment:
I think using `.into_iter()` here would allow us to avoid the `mut` on the `packages` parameter.

This comment also applies to the code for the reverse of this conversion in `parse_lockfile()` which is using `mut` and `drain(..)` instead of `into_iter()`